cpp/tests/groupby/max_tests.cpp

```diff
@@ -459,5 +459,30 @@ TYPED_TEST(groupby_max_floating_point_test, values_with_infinity)
     keys, vals, expected_keys, expected_vals, std::move(agg), force_use_sort_impl::YES);
 }
 
+TYPED_TEST(groupby_max_floating_point_test, values_with_nan)
```

Comment:
Confirm that the added tests would hang forever without the fix code in this PR.